src/main/java/com/redhat/devtools/lsp4ij/features/documentation/LightQuickDocHighlightingHelper.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.redhat.devtools.lsp4ij.features.documentation;
+
+import com.intellij.lang.Language;
+import com.intellij.lang.LanguageParserDefinitions;
+import com.intellij.lang.documentation.DocumentationSettings;
+import com.intellij.openapi.diagnostic.ControlFlowException;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.editor.richcopy.HtmlSyntaxInfoUtil;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.NlsSafe;
+import com.intellij.util.concurrency.annotations.RequiresReadLock;
+import com.intellij.xml.util.XmlStringUtil;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is a copy/paste from JetBrains code
+ * <a href="https://github.com/JetBrains/intellij-community/blob/master/platform/lang-impl/src/com/intellij/lang/documentation/QuickDocHighlightingHelper.kt">QuickDocHighlightingHelper.kt</a>
+ * <p>
+ * translated into Java and which contains only method used to highlight code.

```

Comment:
translated into Java, only containing methods used to highlight code

Reply:
fixed